File: ScaleformUI_Csharp/Menus/UIMenu/SidePanels/UIMenuSidePanel.cs
namespace ScaleformUI.Menu
{
    public enum PanelSide
    {
        Left,
        Right
    }
    public enum SidePanelsTitleType
    {
        Big,
        Small,
        Classic
    }

    public class UIMenuSidePanel
    {
        public virtual bool Selected { get; set; }
        public virtual bool Enabled { get; set; }
        public virtual PanelSide PanelSide { get; set; }
        public virtual void UpdateParent()
        {
        }
        public void SetParentItem(UIMenuItem item)
        {
            ParentItem = item;
        }

        public UIMenuItem ParentItem { get; set; }

    }
}

File: ScaleformUI_V8/src/menus/UIMenu/sidepanels/uimenusidepanel.ts
export enum PanelSide {
    Left,
    Right
}
export enum SidePanelsTitleType {
    Big,
    Small,
    Classic
}

export abstract class UIMenuSidePanel {
    public Selected: boolean = false;
    public Enabled: boolean = true;
    public PanelSide: PanelSide
    public UpdateParent(): void { }
    public SetParentItem(item: UIMenuItem) {
        this.ParentItem = item;
    }

    public ParentItem: UIMenuItem;
}